resources/debugger/components/app.jsx

```diff
@@ -50,32 +51,82 @@ const Tab = styled(NavLink)`
     opacity: 0.5;
     text-decoration: line-through;
   }
-`
+`;
 
 const TabContent = styled.div`
   flex: 1;
   background: white;
   overflow: auto;
   display: flex;
   flex-direction: row;
-`
+`;
 
 const Label = styled.span`
   font-size: 13px;
   text-transform: capitalize;
   font-weight: normal;
   padding-top: 3px;
-`
+`;
 
 const selectedTab = css`
   opacity: 1 !important;
   border-bottom: 1px solid #3d85ee !important;
-`
+`;
 
 class App extends Component {
   constructor(props) {
-    super(props)
-    bridgeHandler(props.dispatch)
+    super(props);
+    bridgeHandler(props.dispatch);
+  }
+
+  componentDidMount() {
+    this.listenForShortcuts();
+  }
+
+  navigateToTab(direction) {
+    var nextTab;
+    var prevTab;
+    var tabs = ReactDOM.findDOMNode(this).getElementsByClassName("tab");
+    for (var i = 0; i < tabs.length; i++) {
+      if (tabs[i].getAttribute("aria-current") == "true") {
+        if (i == 0) {
+          nextTab = i + 1;
+          prevTab = tabs.length - 1;
+        } else if (i == tabs.length - 1) {
+          nextTab = 0;
+          prevTab = i - 1;
+        } else {
+          nextTab = i + 1;
+          prevTab = i - 1;
+        }
+        break;
+      }
+    }
+    if (direction == "next") {
+      tabs[nextTab].click();
+    }
+    if (direction == "prev") {
+      tabs[prevTab].click();
+    }
+  }
```

Comment:
to replace all of this you can create an array to describe the tabs:
```
const tabs = [
{
  url: '/console',
  label: 'Console'
},
...
]
```

then to find the current position you can use
```
const currentTabIndex = tabs.findIndex(t => t.url === this.props.location.pathname)
```
and to go to a new tab:
```
this.props.history.push(tabs[Maty.max(
  Math.min(
    currentTabIndex + (direction === 'next' ? 1 : -1),
    tabs.length - 1
  ),
0)].url)
```

Reply:
Awesome, that worked and it made me learn a couple of things! 😊 👌

Reply:
:+1:

You can also use the array to simplify the render method:
```
<ul>
 {tabs.map(t => <li>...</li>}
</ul>
```

Reply:
Gotcha! Neat and tidy 👍